internal/daemon/worker/session/manager.go

```diff
@@ -113,7 +113,7 @@ func (m *manager) LoadLocalSession(ctx context.Context, id string, workerId stri
 	}
 	// Update the response to the latest
 	actualSess := actualSessRaw.(*sess)
-	actualSess.ApplyLocalStatus(s.GetStatus())
+	actualSess.ApplySessionUpdate(s.resp, s.GetStatus())
```

Comment:
The status reported in `s.GetStatus()` will have just come from `resp.Status` in `newSess`, so while you _can_ pass it independently there isn't really a reason to. You could just fetch it from `resp` as part of the function.

Reply:
Done, thanks!